lib/fluent/plugin/out_splunk-http-eventcollector.rb

```diff
@@ -25,45 +25,72 @@
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 =end
-require 'fluent/mixin/rewrite_tag_name'
+
+# Splunk HTTP Event collector docs
+# http://dev.splunk.com/view/event-collector/SP-CAAAE6M
+# http://docs.splunk.com/Documentation/Splunk/latest/RESTREF/RESTinput#services.2Fcollector
 
 module Fluent
 class SplunkHTTPEventcollectorOutput < BufferedOutput
-  
-  Plugin.register_output('splunk-http-eventcollector', self)
 
-  include Fluent::HandleTagNameMixin
-  include Fluent::Mixin::RewriteTagName
+  Plugin.register_output('splunk-http-eventcollector', self)
 
   config_param :test_mode, :bool, :default => false
-  
+
   config_param :server, :string, :default => 'localhost:8088'
   config_param :verify, :bool, :default => true
   config_param :token, :string, :default => nil
-  
+
   # Event parameters
   config_param :protocol, :string, :default => 'https'
   config_param :host, :string, :default => nil
   config_param :index, :string, :default => 'main'
   config_param :all_items, :bool, :default => false
-  
+
   config_param :sourcetype, :string, :default => 'fluentd'
   config_param :source, :string, :default => nil
   config_param :post_retry_max, :integer, :default => 5
   config_param :post_retry_interval, :integer, :default => 5
-  
+
   # TODO Find better upper limits
   config_param :batch_size_limit, :integer, :default => 262144 # 65535
   #config_param :batch_event_limit, :integer, :default => 100
-  
+
   # Called on class load (class initializer)
   def initialize
     super
     log.trace "splunk-http-eventcollector(initialize) called"
     require 'net/http/persistent'
     require 'openssl'
   end  # initialize
-  
+
+  # Thanks to
+  # https://github.com/kazegusuri/fluent-plugin-prometheus/blob/348c112d/lib/fluent/plugin/prometheus.rb
+  def self.placeholder_expander(log)
+    # Use internal class in order to expand placeholder
+    if defined?(Fluent::Filter) # for v0.12, built-in PlaceholderExpander
+      begin
+        require 'fluent/plugin/filter_record_transformer'
+        if defined?(Fluent::Plugin::RecordTransformerFilter::PlaceholderExpander)
+          # for v0.14
+          return Fluent::Plugin::RecordTransformerFilter::PlaceholderExpander.new(log: log)
+        else
+          # for v0.12
+          return Fluent::RecordTransformerFilter::PlaceholderExpander.new(log: log)
+        end
+      rescue LoadError => e
+        raise ConfigError, "cannot find filter_record_transformer plugin: #{e.message}"
+      end
+    else # for v0.10, use PlaceholderExapander in fluent-plugin-record-reformer plugin
```

Comment:
There's really no point in supporting 0.10 here as the gem itself only specifies 0.12, and would only ever conceivably be updated for 0.14.
```ruby
gem.add_runtime_dependency "fluentd", '~> 0.12'
```